From: Andreas Pehrson <apehrson@mozilla.com>
Date: Tue, 16 May 2023 06:37:00 +0000
Subject: Bug 1832770 - Cherry-pick upstream libwebrtc commit 7b0d7f48fb.
 r=grulja,webrtc-reviewers,mjf

Upstream commit:
    PipeWire capturer: fix fcntl call when duplicating a file descriptor

    The fcntl() call has variable arguments, therefore we need to pass 0 to
    specify there are no other arguments for this call, otherwise we might
    end up with an argument that is random garbage.

    Bug: webrtc:15174
    Change-Id: I34f16a942d80913b667d8ade7eed557b0233be01
    Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/305120
    Reviewed-by: Alexander Cooper <alcooper@chromium.org>
    Commit-Queue: Jan Grulich <grulja@gmail.com>
    Cr-Commit-Position: refs/heads/main@{#40060}

Differential Revision: https://phabricator.services.mozilla.com/D178009
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/47978aa145282f737fc77fcdcadc995e80e2a743
---
 .../desktop_capture/linux/wayland/shared_screencast_stream.cc   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
index 21863b7d8d..17c895088a 100644
--- a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
+++ b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
@@ -450,7 +450,7 @@ bool SharedScreenCastStreamPrivate::StartScreenCastStream(
 
     if (fd >= 0) {
       pw_core_ = pw_context_connect_fd(
-          pw_context_, fcntl(fd, F_DUPFD_CLOEXEC), nullptr, 0);
+          pw_context_, fcntl(fd, F_DUPFD_CLOEXEC, 0), nullptr, 0);
     } else {
       pw_core_ = pw_context_connect(pw_context_, nullptr, 0);
     }
-- 
2.34.1

